### ISSUE TITLE
Interactive Shell: add Code hinting like JS Console

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#1373]
### Short Summary

Basically what the title says.  I suspect that this is going to be a huge pain though. I'm fairly certain that we do very little implementing the various REPLs, in that we simply provide a place for STDIN and STDOUT to be shown.  I think it's still a scintilla view though so hopefully we can port in the Code Intel stuff ontop of it.

**Comment by Naatan** [comment]
yeah this would be a pita, especially given that the console and the interactive shell use completely different technologies. I don't think this is worth the effort it would take. Maybe if ever we rework our REPL integration.

**Comment by cgchoffman** [comment]
I completely disagree.  Why did you close this?  This would be a great tool.  I would use it all the time but I can't.

**Comment by cgchoffman** [comment]
To clarify my intent with this ticket is:
1. add code intel style hinting as you write.  if you've imported a package then you should be able to get functions/classes from it.  If you create a dict (not implicitly through a return type) you should be able to get functions from a dict.  You should get completions for all built-ins
2. This would be for all current interactive shells.

Additional scoping: This should work into a reworking of the interactive shell UX to make them easier to access and use.

**Comment by Naatan** [comment]
In addition to what @cgchoffman said - we need to investigate how to make the REPL more useful and accessible. Carey's main point was that he "would actually use it" if it had proper auto completions. I don't personally feel that auto completions alone will make all the difference, so this ticket should be about making the REPL integration something other than a last resort.
